### PR TITLE
Charles schwab

### PR DESCRIPTION
**Pull request by apurvrathore**

### CODE REVIEW

File: flux_sdk/flux_core/data_models.py [hunk 1]
```diff
@@ -13,6 +27,8 @@ class EmployeeState(Enum):
     ACCEPTED = 3
     TERMINATED = 4
     ACTIVE = 5
+    ON_LEAVE = 6
```
Comment (vguptarippling) [inline flux_sdk/flux_core/data_models.py:30]: EmployeeState containing ON_LEAVE  and MILITARY_LEAVE seems to be too specific for Chales Schwab. Let's use RipplingLeaveTypes to populate for Charles Schwab

File: flux_sdk/flux_core/data_models.py [hunk 2]
```diff
@@ -85,6 +101,10 @@ class Employee:
     is_rehire: bool
     is_international_employee: bool
     martial_status: MaritalStatus
+    leave_of_absence_type: RipplingLeaveTypes
+    leave_of_absence_start_date: datetime
+    leave_of_absence_return_date: datetime
```
Comment (vguptarippling) [inline flux_sdk/flux_core/data_models.py:106]: let's give all the leaves on Employee model 
list[LeaveInfo]
class LeaveInfo
  leave_type
  start_date
  end_date
  is_paid
and let the decision of choosing the appropriate leave be on 3rd party developers

File: flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py [hunk 3]
```diff
@@ -123,6 +123,20 @@ class EoyInfo:
     year_to_date_hours: This dict contains the year to date total hours if the EmploymentType is Hourly
     """
     year_to_date_hours: Decimal
+    """
+    year_to_date_hours: This dict contains the year to date total hours if the EmploymentType is Hourly
+    """
+    year_to_date_pretax_catchup: Decimal
```
Comment (vguptarippling) [inline flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py:129]: is this being calculated on RM ?

File: flux_sdk/flux_core/data_models.py (hunk 2 above)
Comment (vguptarippling) [reply flux_sdk/flux_core/data_models.py:106]: Add the corresponding changes on RM as well?

File: flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py (hunk 3 above)
Comment (apurvrathore) [reply flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py:129]: yes

File: flux_sdk/flux_core/data_models.py [hunk 4]
```diff
@@ -102,6 +116,8 @@ class ContributionType(Enum):
     COMPANY_MATCH_ROTH = 7
     COMPANY_MATCH_AFTER_TAX = 8
     LOAN = 9
+    _401K_CATCHUP = 10
+    _401K_ROTH_CATCHUP = 11
```
Comment (vguptarippling) [inline flux_sdk/flux_core/data_models.py:120]: Let's rename it to ROTH_401K_CATCHUP

File: flux_sdk/flux_core/data_models.py [hunk 5]
```diff
@@ -3,6 +3,20 @@
 from typing import Optional
 
 
+class RipplingLeaveTypes(Enum):
```
Comment (vguptarippling) [inline flux_sdk/flux_core/data_models.py:6]: Let's rename this LeaveType

File: flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py [hunk 6]
```diff
@@ -134,6 +146,14 @@ class EmployeeDeduction:
     amount: Decimal
     total_pay_for_percentage_contribution: Optional[Decimal]
 
+class LeaveInfo:
+    """
+    This contains employee leave of absence details corresponding to a leave type.
+    """
+    leave_type: RipplingLeaveTypes
```
Comment (vguptarippling) [inline flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py:153]: same change reflected here

File: flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py [hunk 7]
```diff
@@ -134,6 +146,14 @@ class EmployeeDeduction:
...
+    start_date: datetime
```
Comment (vguptarippling) [inline flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py:154]: change this to date type

File: flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py [hunk 8]
```diff
@@ -134,6 +146,14 @@ class EmployeeDeduction:
...
+    return_date: datetime
```
Comment (vguptarippling) [inline flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py:155]: same here as well

File: flux_sdk/flux_core/data_models.py [hunk 9]
```diff
@@ -115,6 +131,8 @@ class DeductionType(Enum):
     _401K_LOAN_PAYMENT = 2
     AFTER_TAX_401K = 3
     _403B = 4
+    _401K_CATCHUP = 5
+    ROTH_401K_CATCH = 6
```
Comment (vguptarippling) [inline flux_sdk/flux_core/data_models.py:135]: Change this to ROTH_401K_CATCH

File: flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py [hunk 10]
```diff
@@ -134,6 +146,14 @@ class EmployeeDeduction:
     amount: Decimal
     total_pay_for_percentage_contribution: Optional[Decimal]
 
+class LeaveInfo:
+    """
+    This contains employee leave of absence details corresponding to a leave type.
+    """
+    leave_type: LeaveType
+    start_date: datetime.date
```
Comment (vguptarippling) [inline flux_sdk/pension/capabilities/report_payroll_contributions/data_models.py:154]: change this to date

**Review by vguptarippling — APPROVED** [review]

**Pull request merged**